src/sentry/monitors/endpoints/organization_monitor_details.py

```diff
@@ -114,6 +128,27 @@ def put(self, request: Request, organization, project, monitor) -> Response:
             params["status"] = result["status"]
         if "config" in result:
             params["config"] = result["config"]
+
+            # update timeouts + expected next check-in, as appropriate
+            checkin_margin = result["config"].get("checkin_margin")
+            if checkin_margin != existing_margin:
+                for monitor_environment in MonitorEnvironment.objects.filter(monitor_id=monitor.id):
+                    monitor_environment.next_checkin_latest = (
+                        monitor_environment.next_checkin
+                        + timedelta(minutes=int(checkin_margin or 1))
+                    )
+                    monitor_environment.save(update_fields=["next_checkin_latest"])
+
+            max_runtime = result["config"].get("max_runtime")
+            if max_runtime != existing_max_runtime:
+                for in_progress_checkin in MonitorCheckIn.objects.filter(
+                    monitor_id=monitor.id, status=CheckInStatus.IN_PROGRESS
+                ):
```

Comment:
might be worth copying this to upsert (for older in-progress check-ins) but probably not